OrbitGl/PickingManager.h

```diff
@@ -25,96 +25,99 @@ class Pickable {
   virtual void OnDrag(int /*a_X*/, int /*a_Y*/) {}
   virtual void OnRelease(){};
   virtual void Draw(GlCanvas* a_Canvas, PickingMode a_PickingMode) = 0;
-  virtual bool Draggable() { return false; }
-  virtual bool Movable() { return false; }
-  virtual std::string GetTooltip() const { return ""; }
+  [[nodiscard]] virtual bool Draggable() { return false; }
+  [[nodiscard]] virtual bool Movable() { return false; }
+  [[nodiscard]] virtual std::string GetTooltip() const { return ""; }
+};
+
+enum class PickingType : uint32_t {
+  kInvalid = 0,
+  kLine = 1,
+  kBox = 2,
+  kTriangle = 3,
+  kPickable = 4
 };
 
+// Instances of batchers used to draw must be in 1:1 correspondence with
+// values in the following enum. Currently, two batchers are used, one to draw
+// UI elements (corresponding to BatcherId::UI), and one for drawing events on
+// the time graph (corresponding to BatcherId::kTimeGraph). If you want to add
+// more batchers, this enum must be extended and you need to spend more bits
+// on the batcher_id_ field below. The total number of elements that can be
+// correctly picked is limited to the number of elements that can be encoded
+// in the bits remaining after encoding the batcher id, and the
+// PickingType, so adding more batchers or types has to be carefully
+// considered.
+enum class BatcherId : uint32_t { kTimeGraph, kUi };
+
 struct PickingID {
```

Comment:
While you are renaming, I think that this should become `PickingId`. From the style guide: "For names written in mixed case (also sometimes referred to as "camel case" or "Pascal case"), in which the first letter of each word is capitalized, prefer to capitalize abbreviations as single words, e.g., StartRpc() rather than StartRPC()."

Reply:
Good point, thanks, I renamed this.